override_mail_recipients/models/ir_mail_server.py

```diff
@@ -24,40 +55,79 @@ def patch_message(self, message):
         Odoo. Most likely this message will be catched and an appropiate
         reason for non delivery registered.
         """
-        override_email_id = self.env.ref(
+        override_email_to = self.env.ref(
             'override_mail_recipients.override_email_to',
             raise_if_not_found=False)
-        if not override_email_id or \
-                not override_email_id.value or \
-                override_email_id.value == 'disable':
+        override = override_email_to.value if override_email_to else 'disable'
+        domain_whitelist = self.env.ref(
+            'override_mail_recipients.domain_whitelist',
+            raise_if_not_found=False)
+        whitelisted_domains = [
+            '@' + domain for domain in domain_whitelist.value.split(',')
+            if '.' in domain] if domain_whitelist else []
+        if override == 'disable' and not whitelisted_domains:
             return
-        actual_recipients = extract_rfc2822_addresses(override_email_id.value)
-        assert actual_recipients, 'No valid override_email_to'
+        override_recipients = \
+            extract_rfc2822_addresses(override) \
+            if override != 'disable' else []
+        assert override_recipients or whitelisted_domains, \
+            'No valid override_email_to'
```

Comment:
please add https://github.com/Therp/Therp-Addons/commit/bf6b463ca7c98f84bfab0d0614c3e1f840662340 or something similar to make this not fail on new installations